src/main/java/org/academiadecodigo/bootcamp/scanners/menu/CustomMenuInputScanner.java

```diff
@@ -0,0 +1,36 @@
+package org.academiadecodigo.bootcamp.scanners.menu;
+
+import org.academiadecodigo.bootcamp.scanners.integer.IntegerSetInputScanner;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class CustomMenuInputScanner extends IntegerSetInputScanner {
+    private Map<Integer,String> optionMap;
+
+    public CustomMenuInputScanner(Map<Integer,String> optionMap) {
+        super(optionMap.keySet());
+        this.optionMap = new TreeMap<>(optionMap);
```

Comment:
To be fully customizable, I think it would be nice to let the user also decide the order by which the menuOptions are shown. By using a `TreeMap` you are forcing the options to appear by ascending order. If you use a `LinkedHashMap` for instance, you could let the presentation order to be decided when the optionMap is actually created, and assure that the order is maintained.